stl/inc/algorithm

```diff
@@ -1865,6 +1865,201 @@ _NODISCARD _FwdIt1 find_end(_ExPo&& _Exec, _FwdIt1 _First1, _FwdIt1 _Last1, _Fwd
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // CLASS ranges::_Find_end_fn
+    class _Find_end_fn : private _Not_quite_object {
+    private:
+        template <class _It1, class _It2, class _Pr, class _Pj1, class _Pj2>
+        _NODISCARD static constexpr bool _Equal_rev_pred_unchecked(
+            _It1 _First1, _It2 _First2, const _It2 _Last2, _Pr _Pred, _Pj1 _Proj1, _Pj2 _Proj2) {
+            // compare [_First1, ...) to [_First2, _Last2)
+            constexpr bool _Both_identity = is_same_v<_Pj1, identity> && is_same_v<_Pj2, identity>;
+            // Performance note: teach _Equal_memcmp_is_safe about contiguous iterators and ranges::equal_to
+            if constexpr (_Both_identity && decltype(_Equal_memcmp_is_safe(_First1, _First2, _Pred))::value) {
+#ifdef __cpp_lib_is_constant_evaluated
+                if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+                {
+                    const auto _First1_ch = reinterpret_cast<const char*>(_First1);
+                    const auto _First2_ch = reinterpret_cast<const char*>(_First2);
+                    const auto _Count     = static_cast<size_t>(reinterpret_cast<const char*>(_Last2) - _First2_ch);
+                    return _CSTD memcmp(_First1_ch, _First2_ch, _Count) == 0;
+                }
+            }
+
+            for (; _First2 != _Last2; ++_First1, (void) ++_First2) {
+                if (!_STD invoke(_Pred, _STD invoke(_Proj1, *_First1), _STD invoke(_Proj2, *_First2))) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        template <class _It1, class _It2, class _Pr, class _Pj1, class _Pj2>
+        _NODISCARD static constexpr subrange<_It1> _Random_access_sized(_It1 _First1, _It2 _First2,
+            const iter_difference_t<_It1> _Count1, const iter_difference_t<_It2> _Count2, _Pr _Pred, _Pj1 _Proj1,
+            _Pj2 _Proj2) {
+            if (0 < _Count2 && _Count2 <= _Count1) {
+                const auto _UFirst1 = _Get_unwrapped(_First1);
+                const auto _UFirst2 = _Get_unwrapped(_First2);
+
+                for (auto _UCandidate = _UFirst1 + (_Count1 - static_cast<iter_difference_t<_It1>>(_Count2));;
+                     --_UCandidate) {
+                    if (_Equal_rev_pred_unchecked(_UCandidate, _UFirst2, _UFirst2 + _Count2, _Pass_fn(_Pred),
+                            _Pass_fn(_Proj1), _Pass_fn(_Proj2))) {
+                        _Seek_wrapped(_First1, _UCandidate);
+                        return {_First1, _First1 + static_cast<iter_difference_t<_It1>>(_Count2)};
+                    }
+
+                    if (_UFirst1 == _UCandidate) {

```

Comment:
With `_UCandidate` being the loop variable is this a Yoda condition? 😈

Reply:
I noticed that too, but it's in pre-existing code, so I wasn't sure if it was worth cleaning up both occurrences.

Reply:
I cleaned both.